src/application/look-up-wallet-address/app.tsx

```diff
@@ -0,0 +1,121 @@
+import { FormEvent, useEffect, useState } from 'react';
+import { useDebounce } from 'react-use';
+
+import { useCommandQuery } from 'shared/messaging';
+import { Closable, IconButton } from 'shared/ui';
+import { useExtensionSettings } from 'shared/extension';
+
+import { GetResolvedAddressCommand, GetTwitterIdsCommand } from './commands';
+import { AddressList } from './address-list';
+
+export const App = () => {
+  const [username, setUsername] = useState<string>('');
+  const [inputValue, setInputValue] = useState<string>('');
+  const [visible, setVisible] = useState(false);
+
+  const { extensionSettings } = useExtensionSettings();
+
+  const isEnabled =
+    extensionSettings['entire-extension-enabled'] &&
+    extensionSettings['block-explorers-enabled'];
+
+  const twitterIdsQuery = useCommandQuery({
+    command: new GetTwitterIdsCommand({
+      username,
+    }),
+    staleTime: Number.POSITIVE_INFINITY,
+  });
+
+  const addressesQuery = useCommandQuery({
+    command: new GetResolvedAddressCommand({
+      identifier: username,
+      twitterId: twitterIdsQuery.data?.twitterIDs
+        ? Object.values(twitterIdsQuery.data?.twitterIDs)[0]
+        : undefined,
+    }),
+    staleTime: Number.POSITIVE_INFINITY,
+    enabled: username.length >= 3,
+  });
+
+  useDebounce(
+    () => {
+      setUsername(inputValue);
+    },
+    500,
+    [inputValue],
+  );
+
+  const handleInput = (event: FormEvent<HTMLInputElement>) => {
+    const eventTarget = event.target as HTMLInputElement;
+    setInputValue(eventTarget.value);
+  };
+
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.ctrlKey && event.key === 'i') {
+        setVisible((previous) => {
+          return !previous;
+        });
+      }
+    };
+
+    window.addEventListener('keydown', handleKeyDown);
+
+    return () => {
+      window.removeEventListener('keydown', handleKeyDown);
+    };
+  }, []);
```

Comment:
maybe you can experiment and see if we can use one of these:
https://github.com/streamich/react-use/blob/master/docs/useKey.md
https://github.com/streamich/react-use/blob/master/docs/useKeyPress.md
https://github.com/streamich/react-use/blob/master/docs/useKeyboardJs.md
https://github.com/streamich/react-use/blob/master/docs/useKeyPressEvent.md

Reply:
thanks, the useKey hook works great here